Meta/Lagom/Tools/CodeGenerators/LibLocale/GenerateLocaleData.cpp

```diff
@@ -1375,7 +1376,7 @@ static LanguageMapping const* resolve_likely_subtag(LanguageID const& language_i
             if (!language_id.script.has_value())
                 continue;
 
-            search_key.language = "und"sv;
+            search_key.language = String::from_utf8("und"sv).release_value_but_fixme_should_propagate_errors();
```

Comment:
I was actually thinking about adding something like this to String:
```c++
template<size_t N>
consteval String from_short_string_literal(char const (&string)[N])
{
    static_assert(N <= MAX_SHORT_STRING_BYTE_COUNT);
    return /* create the string object */;
}
```

and then coming back and converting the string literals that fit the bill to this factory

Reply:
SGTM 👍

Reply:
Turns out this doesn't quite work because a `consteval` factory requires the type it returns to be a literal type, which `String` can't be because of its non-constexpr destructor (which it can't be because it's non-trivial).

```
AK/String.h:49:29: error: consteval function's return type 'AK::String' is not a literal type
    static consteval String from_utf8_short_string(StringView string)
                            ^
AK/String.h:41:5: note: 'String' is not literal because its destructor is not constexpr
    ~String();
```

We could do a non-`constexpr` factory like this though:
```c++
static ErrorOr<String> from_utf8(StringView);
static String from_utf8_short_string(StringView);

String String::from_utf8_short_string(StringView view)
{
    VERIFY(view.length() <= MAX_SHORT_STRING_BYTE_COUNT);
    return MUST(from_utf8(view));
}
```